### ISSUE TITLE
Proposal: Add ProfileNavigationViewItem

### ISSUE THREAD

**Issue by ad1Dima** [microsoft/microsoft-ui-xaml#1488]
# Proposal: ProfileNavigationViewItem
Special Item that represents UserProfile
![image](https://user-images.githubusercontent.com/3244664/67453818-1e713e80-f652-11e9-984f-f0f2dfbac125.png)

## Rationale
As you can see in #375, #957 and Microsoft Store app, many apps has dedicated user profile option in side menu. And them looks the same.

Profile has rounded image instead of Icon, so it will be helpful to have ProfileNavigationViewItem with 
PersonPicture inside

``` xml
<Grid x:Name="ContentGrid">
    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="Auto" />
        <ColumnDefinition Width="*" />
    </Grid.ColumnDefinitions>
    <Viewbox 
        x:Name="PictureBox"
        Height="16"
        Width="16"
        Margin="16,0,0,0"
        VerticalAlignment="Center"
        HorizontalAlignment="Center">
        <PersonPicture
            ProfilePicture="{TemplateBinding ProfilePicture}"
            DisplayName="{TemplateBinding DisplayName}"/>
    </Viewbox>
    <ContentPresenter
        x:Name="ContentPresenter"
        Grid.Column="1"
        Margin="8,0,16,0"
        Foreground="{ThemeResource DefaultTextForegroundThemeBrush}"
        TextWrapping="NoWrap"
        ContentTransitions="{TemplateBinding ContentTransitions}"
        ContentTemplate="{TemplateBinding ContentTemplate}"
        Content="{TemplateBinding Content}"
        VerticalAlignment="Center"
        AutomationProperties.AccessibilityView="Raw"/>
</Grid>
```

**Comment by mdtauk** [comment]
Profile as a default item is good, but it may also be worth adding a **NavigationViewButton** which would enable actions which do not navigate, and so won't change the current selected item.  Profile items in this view, may not have a page, but display a flyout.

Another example of this is the Shut Down and Profile items on the Start Screen/Start Menu?

**Comment by ad1Dima** [comment]
> However it may also be worth adding a NavigationViewButton which would enable actions which do not navigate, and so won't change the current selected item.

NavigationViewItem has SelectsOnInvoked property. I guess we don't need dedicated NavigationViewButton, but only NavigationViewItem.Flyout property

**Comment by mdtauk** [comment]
> > However it may also be worth adding a NavigationViewButton which would enable actions which do not navigate, and so won't change the current selected item.
> 
> NavigationViewItem has SelectsOnInvoked property. I guess we don't need dedicated NavigationViewButton, but only NavigationViewItem.Flyout property
>

Would that prevent the selection indicator from moving down to the item clicked?

And specific to a sign in/profile item, there would be a state when logged in, a state while logged out, and the action performed would differ based on that state.

So finding a way to wrap all that into a control (as well as changing based on if its in the side or top orientated NavigationView) would keep things consistent across all apps, instead of expecting devs to go their own way with it.

**Comment by ad1Dima** [comment]
> Would that prevent the selection indicator from moving down to the item clicked?

that is exactly what it does. You can check it [here](https://www.microsoft.com/store/productId/9MSVH128X2ZT)

> So finding a way to wrap all that into a control (as well as changing based on if its in the side or top orientated NavigationView) would keep things consistent across all apps, instead of expecting devs to go their own way with it.

There is too many options. So let developers do what they need. It's not do easy to make you own ProfileNavigationViewItem properly, because of standard template that expect Icon, so i proposed to make it here. But any logic connected with profile and login must be implemented by developers.

**Comment by ad1Dima** [comment]
It seems it must be called NavigationViewItemProfile alongside with NavigationViewItemHeader and NavigationViewItemSeparator

**Comment by ranjeshj** [comment]
Could the profile picture be added as content of the NavigationViewItem and used without setting an Icon (there might be an issue with margins when not using an Icon, but that sounds fixable).

**Comment by ad1Dima** [comment]
> Could the profile picture be added as content of the NavigationViewItem and used without setting an Icon (there might be an issue with margins when not using an Icon, but that sounds fixable).

Yes it could. But I think this is common scenario, why don't we make solution for that?
